drivers/i3c/i3c_npcx.c

```diff
@@ -2057,63 +2459,401 @@ static int npcx_i3c_cntlr_init(const struct device *dev)
 	LOG_DBG("APB4_CLK: %d", apb4_rate);
 
 	if (ret != 0x0) {
-		LOG_ERR("Get APb4 source clock fail %d", ret);
+		LOG_ERR("Get APB4 source clock fail %d", ret);
 		return -EINVAL;
 	}
 
-	bamatch = DIV_ROUND_UP(apb4_rate, MHZ(1));
-	bamatch = DIV_ROUND_UP(bamatch, BAMATCH_DIV);
+	bamatch = get_bus_available_match_val(apb4_rate);
 	LOG_DBG("BAMATCH: %d", bamatch);
 
 	SET_FIELD(inst->CONFIG, NPCX_I3C_CONFIG_BAMATCH, bamatch);
 
 	return 0;
 }
 
+static int npcx_i3c_target_init(const struct device *dev)
+{
+	const struct npcx_i3c_config *config = dev->config;
+	struct npcx_i3c_data *data = dev->data;
+	struct i3c_config_target *config_target = &data->config_target;
+	struct i3c_reg *inst = config->base;
+	const struct device *const clk_dev = config->clock_dev;
+	uint32_t apb4_rate;
+	uint8_t bamatch;
+	int idx_module = GET_MODULE_ID(config->instance_id);
+	int ret;
+	uint64_t pid;
+
+	/* I3C module mdma cotroller or target mode select */
+	npcx_i3c_target_sel(idx_module, true);
+
+	/* Set bus available match value in target register */
+	ret = clock_control_get_rate(clk_dev, (clock_control_subsys_t)&config->ref_clk_subsys,
+				     &apb4_rate);
+	LOG_DBG("APB4_CLK: %d", apb4_rate);
+
+	if (ret != 0x0) {
+		LOG_ERR("Get APB4 source clock fail %d", ret);
+		return -EINVAL;
+	}
+
+	bamatch = get_bus_available_match_val(apb4_rate);
+	LOG_DBG("BAMATCH: %d", bamatch);
+	SET_FIELD(inst->CONFIG, NPCX_I3C_CONFIG_BAMATCH, bamatch);
+
+	/* Enable secondary controller capabillity if supported */
+	if (config->secondary_mode == true) {
+		SET_FIELD(inst->MCONFIG, NPCX_I3C_MCONFIG_CTRENA, MCONFIG_CTRENA_CAPABLE);
+	}
+
+	/* Set Provisional ID */
+	pid = config_target->pid;
+
+	/* PID[47:33] MIPI manufacturer ID */
+	SET_FIELD(inst->VENDORID, NPCX_I3C_VENDORID_VID, (uint32_t)GET_PID_VENDOR_ID(pid));
+
+	/* PID[32] Verdor fixed value(0) or random value(1) */
+	if (config_target->pid_random == true) {
```

Comment:
`	if (config_target->pid_random) {`

pid_random is a bool

Reply:
Done